python-ci/pipeline.yaml

```diff
@@ -0,0 +1,37 @@
+env:
+  PYTHON_VERSION: "3"
+
+steps:
+  - label: ":pip: Install dependencies"
+    key: "pip"
+    command: pip install -r requirements.txt
+    plugins:
+      - docker#v5.9.0:
+          image: "python:${PYTHON_VERSION}"
+
+  - label: Lint with Ruff
+    depends_on: ["pip"]
+    command: |
+      pip install ruff
+      ruff check .
+    plugins:
+      - docker#v5.9.0:
+          image: "python:${PYTHON_VERSION}"
+
+  - label: ":pytest: Run pytest"
+    key: "pytest"
+    depends_on: ["pip"]
+    command: |
+      pip install pytest
+      pytest --junitxml=junit/test-results.xml \
+    artifact_paths:
+      - junit/test-results.xml
+    plugins:
+      - docker#v5.9.0:
+          image: "python:${PYTHON_VERSION}"
+
+  - label: ":junit: Annotate"
+    depends_on: ["pytest"]
```

Comment:
I like the use of array of string(s) here, shows the user you can have multiple dependencies. I think I'll do a sweep of other YAML files to use this too :)

Reply:
Yeah I wouldn't be too phased about it. When running prettier over yaml, I think there's a threshold when it will switch over to multi-line style.